source/ingress.go

```diff
@@ -152,7 +160,11 @@ func endpointsFromIngress(ing *v1beta1.Ingress) []*endpoint.Endpoint {
 			continue
 		}
 
-		endpoints = addEndpointsFromTargetAnnotation(ing, rule.Host, endpoints)
+		endpoints = getEndpointsFromTargetAnnotation(ing, rule.Host)
```

Comment:
@ideahitme My thought was to move it to be called from Endpoint also, but then that leads to duplication of the code that loops over the rules gathering hosts.  

As far as the call to it in the endpointsFromTemplate function, I'm not sure.  If the consensus is to change that I'll be glad to do it, but I'd rather not have this fix get bogged down on something unrelated.